src/compiler/crystal/syntax/lexer.cr

```diff
@@ -2477,7 +2477,40 @@ module Crystal
       end
 
       start = current_pos
-      while !current_char.ascii_whitespace? && current_char != '\0' && current_char != @token.delimiter_state.end
+      sub_start = start
+      value = String::Builder.new
+
+      escaped = false
+      while true
```

Comment:
`loop do`?

Reply:
Everywhere else in this file it's `while true`... I'd rather stick with that until (maybe) all occurances would be converted (not sure if).

Reply:
`loop do` and `while true` are the same. In fact, `loop do` might be a bit slower to compile, and it has a counter.

Please, let's not comment such things, nowhere it says that `loop do` is better than `while true` nor that it's the std/compile style that we prefer.

Reply:
@asterite Since `loop do` is borrowed from Ruby, we could borrow the [style guide rule](https://github.com/bbatsov/ruby-style-guide#infinite-loop) for this too.

Reply:
But they are not the same. That advice is wrong. If you declare a variable inside the `loop`, it won't exist outside it, because of blocks. So I always prefer `while true`. It's also so much lightweight: no block involves, just pure procedural code.

Reply:
So as Ruby's version: https://www.ruby-forum.com/topic/178594#781972

Block version creates a scope, so variables declared inside it won't be leaked.

```crystal
a = 0
loop do
  a += 1
  b = "local"
  break if a > 3
end
puts a # => 4
```

Is there a difference between both on `release` mode?

Reply:
There shouldn't be a different in `release`, but I'm saying that there's a difference in semantics.

Plus I don't care, `loop do` or `while true` do the same, none is better, and none should be favored. Otherwise, let's remove one of the two.

Reply:
Nobody should care about those details, but we are programmers, we fight about ifs on the same line or new line :smile:

I don't think `while true` should be removed, but maybe using a literal in a conditional should be discouraged (`if true`, `when true`, etc). However this PR isn't the place to discuss that.

Reply:
Sure, though coding style SPEC/RFC would help to know what's the accepted stance on certain language constructs.

Reply:
BOTH are accepted. End of discussion.

Reply:
Having a style guide even if it says "both are accepted" would help avoiding such discussions ;)

Reply:
Having no style guide and having everyone stop commenting on such small details would also help.

I mean, we have a formatter already. If it's formatted, we accept it. If there's no performance loss between two constructs, we accept both of them. That's the style guide.

Otherwise, as I said before, let's restrict the language to just the constructs we like. Let's end with a language like Go.

Reply:
Just a comment - the "style guide" in ruby is not official. I don't agree with most of it either. I have no comment on crystal itself, that is up to the crystal devs, but hugoabonizio referred to the style code in ruby, used and created by rubocop + the rubocop devs, and I have to add that this is not an "official" style guide. It's just the default one that the rubocop main dev added and you can change it by customizing rubocop (so I don't have a big problem with it either).

Reply:
Hey, there is [ameba](https://github.com/veelenga/ameba), is a nice tool, is being used on amber framework https://github.com/amberframework/amber/issues/465 and will be probably integrated in scry https://github.com/crystal-lang-tools/scry/issues/58  too :wink: 

[![ameba](https://camo.githubusercontent.com/853cddef3654eca52af6e63976e3a3dd01bd8001/68747470733a2f2f67697463646e2e6c696e6b2f7265706f2f7665656c656e67612f62696e2f6d61737465722f616d6562612f6c6f676f2e706e67)](https://github.com/veelenga/ameba "Try ameba!")

Thank you @veelenga ! :tada: